File: src/main/java/com/project/pantau/mapper/AuthMapper.java
package com.project.pantau.mapper;

import com.project.pantau.dto.auth.AuthResponse;
import com.project.pantau.entity.User;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(
        config = MapStructConfig.class,
        uses = UserMapper.class
)
public interface AuthMapper {
    @Mapping(target = "userResponse", source = "user")
    AuthResponse toResponse(
            String token,
            Long expiresIn,
            User user
    );
}

File: backend/internal/dto/mapper/auth_mapper.go
package mapper

import (
	"pantau/internal/dto/auth"
	"pantau/internal/entity"
)

// AuthToResponse retains a zero-value user DTO when usr is nil because
// AuthResponse.UserResponse is a value, not a pointer.
func AuthToResponse(token string, expiresIn int64, usr *entity.User) *auth.AuthResponse {
	response := &auth.AuthResponse{Token: token, ExpiresIn: expiresIn}
	if userResponse := UserToAuthResponse(usr); userResponse != nil {
		response.UserResponse = *userResponse
	}
	return response
}
